bottlerocket-settings-models/settings-extensions/container-runtime-plugins/src/lib.rs

```diff
@@ -0,0 +1,166 @@
+//! Settings related to Container Runtime Plugins
+use bottlerocket_model_derive::model;
+use bottlerocket_modeled_types::SizeString;
+use bottlerocket_settings_sdk::{GenerateResult, SettingsModel};
+use serde::{Deserialize, Serialize};
+use std::convert::Infallible;
+
+#[model(impl_default = true)]
+pub struct ContainerRuntimePluginsSettingsV1 {
+    soci_snapshotter: SociSnapshotterSettings,
+}
+
+#[model(impl_default = true)]
+pub struct SociSnapshotterSettings {
+    pull_mode: SociSnapshotterPullMode,
+    parallel_pull_unpack: SociSnapshotterParallelPullUnpack,
+}
+
+#[model(impl_default = true)]
+pub struct SociSnapshotterParallelPullUnpack {
+    max_concurrent_downloads: i64,
+    max_concurrent_downloads_per_image: i64,
+    concurrent_download_chunk_size: SizeString,
+    max_concurrent_unpacks: i64,
+    max_concurrent_unpacks_per_image: i64,
+    discard_unpacked_layers: bool,
+}
+
+/// SociSnapshotterPullMode is used to select which pull mode is enabled.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "kebab-case")]
+pub enum SociSnapshotterPullMode {
+    #[default]
+    ParallelPullUnpack,
+}
+
+type Result<T> = std::result::Result<T, Infallible>;
+
+impl SettingsModel for ContainerRuntimePluginsSettingsV1 {
+    type PartialKind = Self;
+    type ErrorKind = Infallible;
+
+    fn get_version() -> &'static str {
+        "v1"
+    }
+
+    fn set(_current_value: Option<Self>, _target: Self) -> Result<()> {
+        // Set anything that can be parsed as ContainerRuntimePluginsSettingsV1.
+        Ok(())
+    }
+
+    fn generate(
+        existing_partial: Option<Self::PartialKind>,
+        _dependent_settings: Option<serde_json::Value>,
+    ) -> Result<GenerateResult<Self::PartialKind, Self>> {
+        Ok(GenerateResult::Complete(
+            existing_partial.unwrap_or_default(),
+        ))
+    }
+
+    fn validate(_value: Self, _validated_settings: Option<serde_json::Value>) -> Result<()> {
+        // ContainerRuntimePluginsSettingsV1 is validated during deserialization.
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use serde_json::json;
+
+    #[test]
+    fn test_generate_container_runtime_plugins_settings() {
+        assert_eq!(
+            ContainerRuntimePluginsSettingsV1::generate(None, None),
+            Ok(GenerateResult::Complete(
+                ContainerRuntimePluginsSettingsV1 {
+                    soci_snapshotter: None,
+                }
+            ))
+        )
+    }
+
+    #[test]
+    fn test_serde_container_runtime_plugins() {
+        let test_json = json!({
+            "soci-snapshotter": {
+                "pull-mode": "parallel-pull-unpack",
+                "parallel-pull-unpack": {
```

Comment:
Seems like this got renamed from "parallel-pull-unpack" to just "parallel" in https://github.com/awslabs/soci-snapshotter/commit/4a78fc699fb41ca6401ac9b457b11bde4014b257.

Reply:
The struct itself was renamed to `Parallel` but the actual TOML is still `parallel_pull_unpack`: https://github.com/awslabs/soci-snapshotter/commit/4a78fc699fb41ca6401ac9b457b11bde4014b257#diff-9c6f48c277a843e85b7c1a21f13bec4adc36b99204b8798bbd0a41ac120800e2R19-R24

we can still choose to model this as just `parallel` if that's preferred for its brevity